components/hal/esp32s3/include/hal/cam_ll.h

```diff
@@ -0,0 +1,638 @@
+ *
+ * @param dev CAM register base address
+ * @param en True to enable, False to bypass
+ */
+static inline void cam_ll_enable_vsync_filter(lcd_cam_dev_t *dev, bool en)
+{
+    dev->cam_ctrl1.cam_vsync_filter_en = en;
+}
+
+/**
+ * @brief Set CAM input data width
+ *
+ * @param dev CAM register base address
+ * @param stride 16: The bit number of input data is 9~16.  8: The bit number of input data is 0~8.
+ */
+static inline void cam_ll_set_input_data_width(lcd_cam_dev_t *dev, uint32_t stride)
+{
+    switch (stride) {
+    case 8:
+        dev->cam_ctrl1.cam_2byte_en = 0;
+        break;
+    case 16:
+        dev->cam_ctrl1.cam_2byte_en = 1;
+        break;
+    default:
+        HAL_ASSERT(false);
+        break;
+    }
+}
+
+/**
+ * @brief Whether to invert CAM_DE
+ *
+ * @param dev CAM register base address
+ * @param en True to invert, False to not invert
+ */
+static inline void cam_ll_enable_invert_de(lcd_cam_dev_t *dev, bool en)
+{
+    dev->cam_ctrl1.cam_de_inv = en;
+}
+
+/**
+ * @brief Whether to invert CAM_HSYNC
+ *
+ * @param dev CAM register base address
+ * @param en True to invert, False to not invert
+ */
+static inline void cam_ll_enable_invert_hsync(lcd_cam_dev_t *dev, bool en)
+{
+    dev->cam_ctrl1.cam_hsync_inv = en;
+}
+
+/**
+ * @brief Whether to invert CAM_VSYNC
+ *
+ * @param dev CAM register base address
+ * @param en True to invert, False to not invert
+ */
+static inline void cam_ll_enable_invert_vsync(lcd_cam_dev_t *dev, bool en)
+{
+    dev->cam_ctrl1.cam_vsync_inv = en;
+}
+
+/**
+ * @brief Enable the mode to control the input control signals
+ *
+ * @param dev CAM register base address
+ * @param mode 1: Input control signals are CAM_DE, CAM_HSYNC and CAM_VSYNC;
+ *             0: Input control signals are CAM_DE and CAM_VSYNC. CAM_HSYNC and CAM_DE are all 1 the the same time.
+ */
+static inline void cam_ll_set_vh_de_mode(lcd_cam_dev_t *dev, bool enable)
+{
+    dev->cam_ctrl1.cam_vh_de_mode_en = enable;
+}
+
+/**
+ * @brief Get the mode of input control signals
+ *
+ * @param dev CAM register base address
+ * @param en The pointer to accept the vh_de mode status. 1: Input control signals are CAM_DE, CAM_HSYNC and CAM_VSYNC;
+ * 0: Input control signals are CAM_DE and CAM_VSYNC. CAM_HSYNC and CAM_DE are all 1 the the same time.
+ */
+static inline void cam_ll_get_vh_de_mode(lcd_cam_dev_t *dev, bool *en)
+{
+    *en = dev->cam_ctrl1.cam_vh_de_mode_en;
+}
+
+/**
+ * @brief Set the wire width of CAM output
+ *
+ * @param dev CAM register base address
+ * @param width CAM output wire width
+ */
+static inline void cam_ll_set_data_wire_width(lcd_cam_dev_t *dev, uint32_t width)
+{
+    // data line width is same as data stride that set in `cam_ll_set_input_data_width`
+}
+
+/**
+ * @brief Start the CAM transaction
+ *
+ * @param dev CAM register base address
+ */
+__attribute__((always_inline))
+static inline void cam_ll_start(lcd_cam_dev_t *dev)
+{
+    dev->cam_ctrl.cam_update = 1;
```

Comment:
Just a hint if something doesn't work out here...
I made the experience that the `cam_update` bit didn't clear itself, which lead to some strange behavior. 
Therefore I manually set it back to 0.